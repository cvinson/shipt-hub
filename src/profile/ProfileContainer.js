import React from 'react';
import PropTypes from 'prop-types';
import Profile from './Profile';
import { getUser, getUserFollowers } from '../services/api';
import './Profile.css';

const ProfileContainer = ({ history, match }) => {
  return (
    <div className="profileContainer">
      <Profile
        className="profileContainer"
        getUser={getUser}
        getUserFollowers={getUserFollowers}
        history={history}
        username={match.params.username} />
    </div>
  );
};

ProfileContainer.propTypes = {
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
}

export default ProfileContainer;
