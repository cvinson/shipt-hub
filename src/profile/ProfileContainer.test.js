import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ProfileContainer from './ProfileContainer';

describe('Profile', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProfileContainer history={{}} match={{ params: { username: 'githubUser' } }} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
