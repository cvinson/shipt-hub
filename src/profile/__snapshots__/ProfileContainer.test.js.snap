// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<div
  className="profileContainer"
>
  <Profile
    className="profileContainer"
    getUser={[Function]}
    getUserFollowers={[Function]}
    history={Object {}}
    username="githubUser"
  />
</div>
`;
